Reduce the size of Dockerfile-centos7 image

jakule opened the pull request:
The Dockerfile for centos7 has been updated to refine the package installation process. The wildcard (*) in the DEVTOOLSET-gcc statement used for 'yum install' command has been removed, targeting a specific version rather than installing any available DEVTOOLSET-gcc packages. It aims to gain more control over the build environment and minimize unintentional side effects.

This path reduces the size from 5.8GB to 4.0GB

jakule committed:
Reduce the size of Dockerfile-centos7 image

The Dockerfile for centos7 has been updated to refine the package installation process. The wildcard (*) in the DEVTOOLSET-gcc statement used for 'yum install' command has been removed, targeting a specific version rather than installing any available DEVTOOLSET-gcc packages. It aims to gain more control over the build environment and minimize unintentional side effects.

reedloden commented on build.assets/Dockerfile-centos7:
```diff
@@ -262,7 +262,7 @@ RUN yum groupinstall -y 'Development Tools' && \
 # As mentioned above, these packages are unsigned.
 RUN yum install --nogpgcheck -y \
-        ${DEVTOOLSET}-gcc* \
+        ${DEVTOOLSET}-gcc    \
```
Unintentional whitespace?

jakule replied:
Removed

reedloden reviewed (commented):
Could you also make the same change to `Dockerfile-centos7-assets` and `Dockerfile-centos7-fips`?

jakule committed:
Refine package installation in Dockerfile-centos7

The DEVTOOLSET-gcc* wildcard in the 'yum install' command has been replaced with DEVTOOLSET-gcc to target a specific version in Dockerfile-centos7 and Dockerfile-centos7-fips. This measure optimizes the build environment, preventing accidental integration of multiple DEVTOOLSET-gcc packages or versions that may inflate the image size.

jakule committed:
Add DEVTOOLSET-gcc-c++ to centos Dockerfiles

The Dockerfiles for both centos7 and centos7-fips have been updated to install the DEVTOOLSET-gcc-c++ package. Specifying the gcc-c++ package directly, instead of using the wildcard, provides more precise control over the build environment and prevents undesired packages from being installed and inflating the docker image size.
